app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -464,6 +464,25 @@ def validate(self, data):
                     )
                 }
             )
+
+        if self.instance and "targeting_config_slug" in data:
+            targeting_config_slug = data["targeting_config_slug"]
+            application_choice = next(
+                choice
+                for choice in NimbusExperiment.Application
+                if choice.value == self.instance.application
+            )
```

Comment:
Oh thank you for adding!  You can look up the name of an enum from the value like this:

```py
In [7]: experiment.Application(experiment.application).name
Out[7]: 'FENIX'
```

Reply:
Oh hey that worked, I was trying `[]` subscript notation and getattr and getting nowhere